doc/source/serve/autoscaling-guide.md

```diff
@@ -0,0 +1,369 @@
+(serve-autoscaling)=
+
+# Autoscaling
+
+## Scaling horizontally with `num_replicas`
+
+One way to scale out your deployment without using autoscaling is by increasing the number of replicas. Each deployment consists of one or more [replicas](serve-architecture-high-level-view).
+The number of replicas is specified by the `num_replicas` field in the deployment options.
+By default, `num_replicas` is 1.
+
+```{literalinclude} doc_code/managing_deployments.py
+:start-after: __scaling_out_start__
+:end-before: __scaling_out_end__
+:language: python
+```
+
+## Autoscaling Important Concepts
+
+Instead of setting a fixed number of replicas for a deployment, you can configure a deployment to autoscale based on incoming traffic. The Serve autoscaler adjusts to traffic spikes by observing queue sizes and making scaling decisions to add or remove replicas. To configure it, you can set the `autoscaling_config` field in deployment options.
+
+:::{note}
+The Ray Serve Autoscaler is an application-level autoscaler that sits on top of the [Ray Autoscaler](cluster-index).
+Concretely, this means that the Ray Serve autoscaler asks Ray to start a number of replica actors based on the request demand.
+If the Ray Autoscaler determines there aren't enough available CPUs to place these actors, it responds by requesting more Ray nodes.
+The underlying cloud provider will then respond by adding more nodes.
+Similarly, when Ray Serve scales down and terminates some replica actors, it will try to do so in a way that results in the most nodes having no Ray actors or tasks running on them, at which point the Ray autoscaler will remove those nodes.
```

Comment:
```suggestion
Similarly, when Ray Serve scales down and terminates replica actors, it will attempt to make as many nodes idle as possible so the Ray autoscaler can remove them.
```